### ISSUE TITLE
"Insufficient permissions. Select 'Retry as Sudo' to retry as superuser"

### ISSUE THREAD

**Issue by sneko** [microsoft/vscode#134506]
Does this issue occur when all extensions are disabled?: (it's rare and random... not able to confirm disabling extensions works)

- VS Code Version: `v1.60.2`
- OS Version: `macOS v11.6`

Steps to Reproduce:

Hi,

I'm sorry not being able to provide much detail but:

I'm using VSCode for years and since a few months I got 2 or 3 times a weird issue: I use VSCode as usual, and randomly when I want to save one of my file I get:
```
Insufficient permissions. Select 'Retry as Sudo' to retry as superuser
```

When noticing this, I'm unable save any file, to browse the file tree (it disappear), my only solution is to copy/paste all the unsaved content for backup.

Then to fix it:
* either by setting a new time chown on the current folder https://askubuntu.com/a/1151270
* or by unchecking/checking "Documents folder" as on the image
![image](https://user-images.githubusercontent.com/8195958/136224501-d237c6a9-0fb6-47be-af8b-fbb83b24dc06.png)

It makes no sense to me since I'm not manipulating my folders/files rights 😢 

Next time I won't forget to do a `ls -l .` to see if anything has changed (I might have done that months ago but I don't remember, that's why I open a issue too, to track).

I searched for similar issues on this repo but it seems not "random and rare" for other people.

Note: I wonder if MacOS using `zsh` could be something responsible for this? Could it impact VSCode?

Have anyone of you faced this issue once?

Thank you,

**Comment by sneko** [comment]
@bpasero I got this a new time...

I tried directly to do `ls -l .` from the VSCode terminal but got `ls: .: Operation not permitted`. I checked `whoami` and was with the right user.

Once more I check/uncheck immediately a setting and it works without relaunching nor closing any VSCode terminal (and able too to save files again)
![image](https://user-images.githubusercontent.com/8195958/136254647-d6983157-1813-4f9d-a8b9-cf329b70a36f.png)

I'm really lost haha... hope someone has faced this already ^^...

**Comment by bpasero** [comment]
I do not think this is something VS Code can resolve, so I am inclined to close this. It sounds you should maybe seek support on StackOverflow or other places to ask why the operating system is suddenly to not accept you as a user with permissions anymore.
